src/Compilers/CSharp/Portable/Symbols/Tuples/TupleErrorFieldSymbol.cs

```diff
@@ -0,0 +1,122 @@
+        /// Otherwise, (-1 - [index in members array]);
+        /// </summary>
+        private readonly int _tupleFieldId;
+
+        private readonly ImmutableArray<Location> _locations;
+        private readonly DiagnosticInfo _useSiteDiagnosticInfo;
+
+        public TupleErrorFieldSymbol(NamedTypeSymbol container, string name, int tupleFieldId, Location location, TypeSymbol type, DiagnosticInfo useSiteDiagnosticInfo)
+            : base(container, name, isPublic:true, isReadOnly:false, isStatic:false)
+        {
+            Debug.Assert(name != null);
+            _type = type;
+            _locations = location == null ? ImmutableArray<Location>.Empty : ImmutableArray.Create(location);
+            _useSiteDiagnosticInfo = useSiteDiagnosticInfo;
+            _tupleFieldId = tupleFieldId;
+        }
+
+        /// <summary>
+        /// If this field represents a tuple element (including the name match), 
+        /// id is an index of the element (zero-based).
+        /// Otherwise, (-1 - [index in members array]);
+        /// </summary>
+        public override bool IsTupleField
+        {
+            get
+            {
+                return true;
+            }
+        }
+
+        public override FieldSymbol TupleUnderlyingField
+        {
+            get
+            {
+                // Failed to find one
+                return null;
+            }
+        }
+
+        public override ImmutableArray<Location> Locations
+        {
+            get
+            {
+                return _locations;
+            }
+        }
+
+        public override ImmutableArray<SyntaxReference> DeclaringSyntaxReferences
+        {
+            get
+            {
+                return GetDeclaringSyntaxReferenceHelper<CSharpSyntaxNode>(_locations);
+            }
+        }
+
+        public override bool IsImplicitlyDeclared
+        {
+            get
+            {
+                return false;
+            }
+        }
+
+        internal override bool SuppressDynamicAttribute
+        {
+            get
+            {
+                return true;
+            }
+        }
+
+        internal override TypeSymbol GetFieldType(ConsList<FieldSymbol> fieldsBeingBound)
+        {
+            return _type;
+        }
+
+        internal override DiagnosticInfo GetUseSiteDiagnostic()
+        {
+            return _useSiteDiagnosticInfo;
+        }
+
+        public override sealed int GetHashCode()
+        {
+            return Hash.Combine(ContainingType.GetHashCode(), _tupleFieldId.GetHashCode());
+        }
+
+        public override bool Equals(object obj)
+        {
+            return Equals(obj as TupleErrorFieldSymbol);
+        }
+
+        public bool Equals(TupleErrorFieldSymbol other)
+        {
+            if ((object)other == this)
+            {
+                return true;
+            }
+
+            return (object)other != null && _tupleFieldId == other._tupleFieldId && ContainingType == other.ContainingType;
```

Comment:
Names are factored into equality of containing types. There is no reason to compare names explicitly here.